File: implementations/all-swift/AllTodo/Data/TodoEntity.swift
import Foundation
import SwiftData

@Model
final class TodoEntity {
    @Attribute(.unique) var id: String
    var title: String
    var memo: String?
    var isDone: Bool
    var sortOrder: Int
    var createdAt: Int64
    var updatedAt: Int64

    init(
        id: String,
        title: String,
        memo: String?,
        isDone: Bool,
        sortOrder: Int,
        createdAt: Int64,
        updatedAt: Int64
    ) {
        self.id = id
        self.title = title
        self.memo = memo
        self.isDone = isDone
        self.sortOrder = sortOrder
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}

extension TodoEntity {
    convenience init(item: TodoItem) {
        self.init(
            id: item.id,
            title: item.title,
            memo: item.memo,
            isDone: item.isDone,
            sortOrder: item.sortOrder,
            createdAt: item.createdAt,
            updatedAt: item.updatedAt
        )
    }

    func toDomain() -> TodoItem {
        TodoItem(
            id: id,
            title: title,
            memo: memo,
            isDone: isDone,
            sortOrder: sortOrder,
            createdAt: createdAt,
            updatedAt: updatedAt
        )
    }

    func apply(_ item: TodoItem) {
        title = item.title
        memo = item.memo
        isDone = item.isDone
        sortOrder = item.sortOrder
        createdAt = item.createdAt
        updatedAt = item.updatedAt
    }
}

File: implementations/all-kotlin/app/src/main/java/com/example/crosstodo/allkotlin/data/TodoEntity.kt
package com.example.crosstodo.allkotlin.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Room persistence model. Column names follow the shared SQL schema
 * (snake_case); property names stay camelCase.
 */
@Entity(tableName = "todo_item")
data class TodoEntity(
    @PrimaryKey val id: String,
    @ColumnInfo(name = "title") val title: String,
    @ColumnInfo(name = "memo") val memo: String?,
    @ColumnInfo(name = "is_done") val isDone: Boolean,
    @ColumnInfo(name = "sort_order") val sortOrder: Int,
    @ColumnInfo(name = "created_at") val createdAt: Long,
    @ColumnInfo(name = "updated_at") val updatedAt: Long,
)

fun TodoEntity.toDomain(): TodoItem = TodoItem(
    id = id,
    title = title,
    memo = memo,
    isDone = isDone,
    sortOrder = sortOrder,
    createdAt = createdAt,
    updatedAt = updatedAt,
)

fun TodoItem.toEntity(): TodoEntity = TodoEntity(
    id = id,
    title = title,
    memo = memo,
    isDone = isDone,
    sortOrder = sortOrder,
    createdAt = createdAt,
    updatedAt = updatedAt,
)
